COM-256: Add State/Municipality search autocomplete functionality

ijoel opened the pull request:
PYR1-256: Improve statistics chart rendering

## Purpose
Adds an autocomplete feature to the existing state and municipality search. As the user types, the state/municipality lists are filtered to more quickly navigate to desired coordinates.

![image](https://github.com/sig-gis/comimo/assets/1130619/acab562b-0887-4f6c-994d-d3c7aad595d2)

sifbiri commented:
Hello Joel, COM-256 specifies that we should update the "Validate" panel and not the "Search" or "Subscribe" panels. However, that looks useful. You'll need to add a search/filter feature to the "Validate" panel so users can select and filter through custom municipalities when they try to create a project.

sifbiri commented:
That looks good. However, I have some concerns about the "select" functionality in both "Search" and "Subscribe". Given the icon and the instruction to "select", the user expects a dropdown list, but the field actually performs "autocomplete". So, if the user selects a state and later decides to choose a different one, pressing the "dropdown" icon next to it does nothing (since it has already completed). I believe this is a bit unintuitive. What do you think? Also, there's extra vertical spacing between the two input fields (state and municipality).

ijoel commented:
You should be able to change your selection after completing an entry. This could just be a bug. Thanks for noting the problem.

ijoel commented:
I made some adjustments by resetting the suggestions list on dropdown
click
I hooked into the autocomplete dropdown click event, to attempt to
manually reset completed field value. This works in some instances
doesn't catch them all.

The autocomplete component from "Primereact" is well documented but
because of the applications broad dependence on React 16, I was unable
to install anything newer than version 4.2.2 of the Primereact
autocomplete module.

Since the updates that were made to Search.jsx are ancillary to the
focus of ticket COM-256, I'm going to remove the "Primereact
Autocomplete" and attempt retain some the improvements that were made.

sifbiri commented on src/js/components/Search.jsx:
```diff
@@ -124,95 +185,129 @@ export default function Search({ theMap, featureNames, mapquestKey, isPanel }) {
+                }}
+                onDropdownClick={(e) => {
+                  const query = e.query;
+                  console.log(inputStateValue);
```
Should be deleted

sifbiri commented:
Thank you, Joel. I tried it, and selecting the state works the first time. However, I can't change it afterwards; clicking on the dropdown doesn't bring up the list selection. While it's not a major issue, the user will need to clear the input or type something new for the dropdown to function again. Separately, in the "Search" panel, after a user selects or completes the state and municipality entries, it doesn't focus or trigger the search on the map as intended.

ijoel commented:
Nice, thanks for sharing the gif.  So I was trying to share, on previous comment, that the Primereact autocomplete was just not going to provide the desired functionality for this task.

I removed it, and refactored the code to reuse the existing `react-select` dropdown.  Please let me know you run into any issues.

![image](https://github.com/sig-gis/comimo/assets/1130619/b7568f3a-1836-4b6a-aa34-0cc871378e08)

sifbiri commented:
Thank you, Joel. It looks like it's working again. It would be great if you could fix the spacing between "Select a state" and "Select a municipality" before merging, to match how it currently is in production.
<img width="464" alt="image" src="https://github.com/sig-gis/comimo/assets/8908139/5f7e7fd0-0efa-4ce1-8d55-e8d6c7fa5fda">

ijoel commented:
I removed some padding. I prefer the added space, but it does not more closely match production. 

![image](https://github.com/sig-gis/comimo/assets/1130619/b50336b2-528a-44d8-a94b-50906da6840a)
